src/library/ButtonGroup/ButtonGroup.js

```diff
@@ -0,0 +1,293 @@
+        borderRightColor: rtl && theme.ButtonGroupButton_borderStartColor
+      }
+    }
+  };
+};
+
+const isChecked = (checked: number | Array<number> | Set<number>, index) => {
+  const isSet = checked instanceof Set;
+  const checkedSet = isSet ? checked : setFromArray(toArray(checked));
+  // $FlowFixMe - Refinement to Set not working
+  return checkedSet.has(index);
+};
+
+const Root = createStyledComponent('div', styles, {
+  displayName: 'ButtonGroup',
+  includeStyleReset: true
+});
+
+const getDefaultCheckedState = (props: Props) => {
+  const { children: _children, defaultChecked, mode } = props;
+  const children = Children.toArray(_children);
+
+  if (mode && defaultChecked !== undefined) {
+    const defaultCheckedArray = toArray(defaultChecked);
+    const checked =
+      mode === 'checkbox' ? defaultCheckedArray : [defaultCheckedArray[0]];
+    return setFromArray(checked);
+  }
+
+  const checked = new Set();
+
+  children.map((child, index) => {
+    if (mode === 'checkbox') {
+      if (child.props.defaultChecked) {
+        checked.add(index);
+      }
+    } else if (mode === 'radio') {
+      const selectedChild = children.find(
+        (child) => child.props.defaultChecked
+      );
+      const index = children.indexOf(selectedChild);
+      if (index !== -1) {
+        checked.add(index);
+      }
+    }
+  });
+
+  return checked;
+};
+
+/**
+ * ButtonGroup allows authors to stylistically group a set of related
+ * [Buttons](/components/button) or construct a group of selectable Buttons that
+ * behave like [Radios](/components/radio) or [Checkboxes](/components/checkbox).
+ */
+export default class ButtonGroup extends Component<Props, State> {
+  static displayName = 'ButtonGroup';
+
+  state: State = {
+    checked: getDefaultCheckedState(this.props)
+  };
+
+  render() {
+    const {
+      children,
+      disabled,
+      fullWidth,
+      mode,
+      onClick: ignoreOnClick,
+      size,
+      variant,
+      ...restProps
+    } = this.props;
+    const rootProps = {
+      disabled,
+      fullWidth,
+      mode,
+      role: mode === 'radio' ? 'radiogroup' : 'group',
+      ...restProps
+    };
+    const checked = this.getControllableValue('checked');
+    const buttons = Children.map(children, (child, index) => {
+      const isChildToggleable = mode;
+      const isChildChecked = isChecked(checked, index);
+
+      return cloneElement(child, {
+        ...(isChildToggleable ? { 'aria-checked': isChildChecked } : undefined),
+        'data-index': index,
+        disabled: disabled || child.props.disabled,
+        key: index,
+        ...(isChildToggleable && isChildChecked
+          ? { primary: true }
+          : undefined),
+        onClick: composeEventHandlers(
+          child.props.onClick,
+          this.handleClick.bind(null, index)
+        ),
+        ...(isChildToggleable ? { role: mode } : undefined),
+        size,
+        ...(variant ? { variant } : undefined)
```

Comment:
Isn't this just `variant`?

Reply:
It is, except writing it this way prevents passing the prop at all, whereas if variant is defined as `undefined`, I believe it is still passed to Button.  Correct me if I'm wrong please!

Reply:
I believe that you are correct Victoria.